apps/pie-storybook/stories/pie-button.stories.ts

```diff
@@ -120,91 +120,59 @@ const Template = ({
         </pie-button>
         `;
 
-export const Primary: Story<ButtonProps> = (args: ButtonProps) => Template(args);
-Primary.args = {
-    ...defaultArgs,
-};
+interface ButtonStoryOptions extends StoryOptions {
+    variant?: Variant;
+}
 
-export const Secondary: Story<ButtonProps> = (args: ButtonProps) => Template(args);
-Secondary.args = {
-    ...defaultArgs,
+const createStory = <T>(opts?: ButtonStoryOptions) => ({
+    render: (args: T) => Template(args),
+    args: {
+        ...defaultArgs,
+        ...(opts && opts.variant ? { variant: opts.variant } : {}),
```

Comment:
I just realised this is generic but has a button-specific code block (about variants) - we should probably not do this and find a make to make it properly generic. 

Otherwise I would take this idea (as I really like it) and move it to a ticket and use this code as a starting point :)

Reply:
Yeh, I was going to abstract this out as a follow-up task. For now, it's slightly button-specific until I move it out into it's own method.

Reply:
Ticket for this work is DSW-1215.